sandbox/modules/identity/src/main/java/org/opensearch/identity/SecurityFilter.java

```diff
@@ -67,6 +72,26 @@ private <Request extends ActionRequest, Response extends ActionResponse> void ap
         ActionListener<Response> listener,
         ActionFilterChain<Request, Response> chain
     ) {
+        // TODO The section below shows permission check working for index create and index search, but needs
+        // expanded to work for all actions
+        Subject currentSubject = Identity.getAuthManager().getSubject();
+        // can be multiple index patterns
+        Optional<String[]> resources = Optional.empty();
+        if (request instanceof IndicesRequest) {
+            String[] indices = ((IndicesRequest) request).indices();
+            if (indices != null) {
+                resources = Optional.of(((IndicesRequest) request).indices());
+            }
+        }
+        String permission = action;
+        if (resources.isPresent()) {
+            permission += "|" + Strings.join(resources.get(), ",");
+        }
+
+        if (log.isDebugEnabled()) {
```

Comment:
Everything is still being granted, but this block logs out the result of privilege evaluation.

Add `logger.org.opensearch.identity: debug` to `opensearch.yml` to enable debug logging for this module.